src/System.Text.Primitives/System/Text/Formatting/Internal/TimeFormatter.Utf8.cs

```diff
@@ -0,0 +1,315 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    internal static class InvariantUtf8TimeFormatter
```

Comment:
They are fundamentally different enough that you can't merge them without using generics or some level of indirection since they operate using different sized characters and types. The point of separating them was to avoid indirection and abstraction.